VultisigApp/VultisigApp/Services/AddressService.swift

```diff
@@ -0,0 +1,89 @@
+//
+//  AddressService.swift
+//  VultisigApp
+//
+//  Created by Enrique Souza Soares on 06/08/24.
+//
+
+import Foundation
+import SwiftUI
+import WalletCore
+
+public class AddressService: ObservableObject {
+    
+    static let shared = AddressService()
+    
+    init(){}
+    
+    func resolveDomaninAddress(address: String, chain: Chain) async -> String {
+        
+        do {
+            
+            guard address.isNameService() else {
+                return address
+            }
+            
+            let ensName = address
+            let namehash = ensName.namehash()
+            print("Namehash for \(ensName): \(namehash)")
+            
+            let factory = try EvmServiceFactory.getService(forChain: chain)
+            let address = try await factory.resolveENS(ensName: ensName)
+            
+            print("Resolved address \(address)")
+            
+            return address
+            
+        } catch {
+            
+            print("Error to extract the DOMAIN ADDRESS: \(error.localizedDescription)")
+            return address
+            
+        }
+    }
+    
+    func validateAddress(address: String, chain: Chain) -> Bool {
```

Comment:
This method is misleading to say at least , `isNameService()` will cause this method to by pass the validation as a whole ,  even something fit the `nameService` pattern , it doesn't mean it is a valid address. 

Instead , I think we should resolve the name service , and then perform address validation

Reply:
I only resolve it on the validated form; otherwise, I will need to make them all async, and we will make more calls to RPC.